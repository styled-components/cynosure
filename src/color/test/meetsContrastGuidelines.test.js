// @flow
import meetsContrastGuidelines from '../meetsContrastGuidelines'

describe('meetsContrastGuidelines', () => {
  it('should return the color contrast guidelines of two hex colors', () => {
    expect(meetsContrastGuidelines('#444', '#fff')).toMatchSnapshot()
  })

  it('should return the color contrast guidelines of two 8-digit hex colors', () => {
    expect(meetsContrastGuidelines('#6564CDB3', '#ffffff')).toMatchSnapshot()
  })

  it('should return the color contrast guidelines of two 4-digit hex colors', () => {
    expect(meetsContrastGuidelines('#0f08', '#000')).toMatchSnapshot()
  })

  it('should return the color contrast guidelines of two rgba colors', () => {
    expect(
      meetsContrastGuidelines('rgba(101,100,205,0.7)', 'rgba(0,0,0,1)'),
    ).toMatchSnapshot()
  })

  it('should return the color contrast guidelines of two rgb colors', () => {
    expect(
      meetsContrastGuidelines('rgb(204,205,100)', 'rgb(0,0,0)'),
    ).toMatchSnapshot()
  })

  it('should return the color contrast guidelines of two hsla colors', () => {
    expect(
      meetsContrastGuidelines(
        'hsla(250, 100%, 50%, 0.2)',
        'hsla(0, 100%, 100%, 1)',
      ),
    ).toMatchSnapshot()
  })

  it('should return the color contrast guidelines of two hsl colors', () => {
    expect(
      meetsContrastGuidelines('hsl(0, 100%, 50%)', 'hsl(0, 100%, 100%)'),
    ).toMatchSnapshot()
  })

  it('should return the color contrast guidelines of two named CSS colors', () => {
    expect(meetsContrastGuidelines('papayawhip', 'black')).toMatchSnapshot()
  })

  it('should return failing contrast guidelines when used with a transparent color', () => {
    expect(meetsContrastGuidelines('transparent', '#000')).toMatchSnapshot()
  })
})
