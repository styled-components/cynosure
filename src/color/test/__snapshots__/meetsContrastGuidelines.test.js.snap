// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meetsContrastGuidelines should return failing contrast guidelines when used with a transparent color 1`] = `
Object {
  "AA": false,
  "AAA": false,
  "AAALarge": false,
  "AALarge": false,
}
`;

exports[`meetsContrastGuidelines should return the color contrast guidelines of two 4-digit hex colors 1`] = `
Object {
  "AA": true,
  "AAA": true,
  "AAALarge": true,
  "AALarge": true,
}
`;

exports[`meetsContrastGuidelines should return the color contrast guidelines of two 8-digit hex colors 1`] = `
Object {
  "AA": true,
  "AAA": false,
  "AAALarge": true,
  "AALarge": true,
}
`;

exports[`meetsContrastGuidelines should return the color contrast guidelines of two hex colors 1`] = `
Object {
  "AA": true,
  "AAA": true,
  "AAALarge": true,
  "AALarge": true,
}
`;

exports[`meetsContrastGuidelines should return the color contrast guidelines of two hsl colors 1`] = `
Object {
  "AA": false,
  "AAA": false,
  "AAALarge": false,
  "AALarge": true,
}
`;

exports[`meetsContrastGuidelines should return the color contrast guidelines of two hsla colors 1`] = `
Object {
  "AA": true,
  "AAA": true,
  "AAALarge": true,
  "AALarge": true,
}
`;

exports[`meetsContrastGuidelines should return the color contrast guidelines of two named CSS colors 1`] = `
Object {
  "AA": true,
  "AAA": true,
  "AAALarge": true,
  "AALarge": true,
}
`;

exports[`meetsContrastGuidelines should return the color contrast guidelines of two rgb colors 1`] = `
Object {
  "AA": true,
  "AAA": true,
  "AAALarge": true,
  "AALarge": true,
}
`;

exports[`meetsContrastGuidelines should return the color contrast guidelines of two rgba colors 1`] = `
Object {
  "AA": false,
  "AAA": false,
  "AAALarge": false,
  "AALarge": true,
}
`;
